charts/replicated-library/templates/lib/_ingress.tpl

```diff
@@ -50,7 +50,11 @@ spec:
           {{- $service := $serviceName -}}
           {{- $port := 80 -}}
           {{- if .service -}}
-            {{- $service = default $service .service.name -}}
+            {{- $service = default $.ContextNames.ingress $values.serviceName -}}
+              {{- if .service.name -}}
+             {{- $service = .service.name -}}
+            {{- end }}
+             {{- $service = printf "%s-%s" (include "replicated-library.names.prefix" $) $service -}}
```

Comment:
Similar to https://github.com/replicatedhq/helm-charts/blob/main/charts/replicated-library/templates/lib/_pod_annotations.tpl#L40, you'll want to make sure you `trim` and `truncate` here.

Reply:
Thanks for the feedback.